src/sql-lexer/src/lexer.rs

```diff
@@ -130,7 +152,47 @@ macro_rules! bail {
 ///
 /// See the module documentation for more information about the lexical
 /// structure of SQL.
-pub fn lex(query: &str) -> Result<Vec<(Token, usize)>, LexerError> {
+#[cfg(not(target_arch = "wasm32"))]
+pub fn lex(query: &str) -> Result<Vec<PosToken>, LexerError> {
+    lex_inner(query)
+}
+
+#[cfg(target_arch = "wasm32")]
+#[wasm_bindgen(js_name = PosToken, getter_with_clone, inspectable)]
+#[derive(Debug)]
+pub struct JsToken {
```

Comment:
Can you not just slap the `wasm_bindgen` on the `PosToken`?

Reply:
Oh, I see, `Token` vs `String`, and `wasm_bindgen` probably would choke on the `enum`. Makes sense.

Reply:
Yep. I started trying to plumb this through but stopped because it was getting pretty invasive (`wasm_bindgen` works best with C-like enums). I think a more comprehensive solution would be to bring in [another library like serde-wasm-bindgen](https://github.com/cloudflare/serde-wasm-bindgen), but that would bloat package size for what was easily solved with another struct.